File: node-express/app/Controllers/Controller.js
const Country = require('../Models/Country');
const User = require('../Models/User');

class Controller 
{
	hello(req, res) {
		return res.json({ hello: 'world' });
	}

	static isPrime(n) {
		var result = true;
		
		if (n <= 1) {
			result = false;
		}
		else {
			// We only need to check up to the sqrt of the number.
			for (var i=2; i*i<=n; i++) {
				if (n % i === 0) {
					// This number is evenly divisible by i, so it is not prime.
					result = false;
					break;
				}
			}
		}
		
		return result;
	}
	
	/**
	 * @param {number} n
	 * @return {number}
	 */
	static countPrimes(n) {
		var result = 0;
		
		for (var i=0; i<n; i++) {
			if (Controller.isPrime(i)) {
				result++;
			}
		}
		
		return result;
	}

	static isPrimeInt(n) {
		var result = true;
		let tUint = new Uint32Array(1);
		tUint[0] = n;
		if (tUint[0] <= 1) {
			result = false;
		}
		else {
			// We only need to check up to the sqrt of the number.
			for (var i=2; i*i<=tUint[0]; i++) {
				if (tUint[0] % i === 0) {
					// This number is evenly divisible by i, so it is not prime.
					result = false;
					break;
				}
			}
		}
		
		return result;
	}
	
	/**
	 * @param {number} n
	 * @return {number}
	 */
	static countPrimesInt(n) {
		let result = new Uint32Array(1);
		
		for (var i=0; i<n; i++) {
			if (Controller.isPrimeInt(i)) {
				result[0]++;
			}
		}
		
		return result[0];
	}

	compute(req, res) {
		let x = 0, y = 1;

		let max = 10000 + Math.random() * 500;

		for (let i = 0; i <= max; i++) {
		    let z = x + y;
		    x = y;
		    y = z;
		}

		return res.json({ status: 'done' })
	}

	prime(req, res) {
		var i = Controller.countPrimes(10000000);

		return res.json({ status: 'done' , result:i})
	}

	primeInt(req, res) {
		var i = Controller.countPrimesInt(10000000);

		return res.json({ status: 'done' , result:i})
	}

	async countries(req, res) {
		let data = await Country.fetchAll();

		return res.json({ data });
	}

	async users(req, res) {
		let data = await User.query(q => {
				q.innerJoin('UserCountryMapping', 'User.id', 'UserCountryMapping.userId');
				q.innerJoin('Country', 'UserCountryMapping.countryId', 'Country.id');
				q.groupBy('User.id');
				q.where('UserCountryMapping.countryId', '17');
			})
			.fetchAll({
			 	withRelated: ['countries']
			})

		return res.json({ data });
	}
}

module.exports = new Controller();

File: java-spring/src/main/java/app/Controller.java
package app;

import java.util.List;
import java.util.Random;

import org.hibernate.SessionFactory;
import org.hibernate.criterion.Restrictions;
import org.json.JSONException;
import org.json.JSONObject;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import model.Country;
import model.User;

@RestController
public class Controller {
	
	final private SessionFactory sessionFactory;
	
	public Controller(SessionFactory sessionFactory) {
        this.sessionFactory = sessionFactory;
    }

    @RequestMapping(value = "/hello", produces = "application/json")
    public String hello() throws JSONException {
        return new JSONObject().put("hello", "world").toString();
    }
    
    @RequestMapping(value = "/compute", produces = "application/json")
    public String compute() throws JSONException {
        long x = 0, y = 1, z, max;
        
        Random r = new Random();
        max = 10000 + r.nextInt(500);
        
        for (int i = 0; i <= max; i++) {
        	z = x + y;
		    x = y;
		    y = z;
		}
        
        return new JSONObject().put("status", "done").toString();
    }
    
    public boolean isPrime(long n) {
        boolean result = true;
    
        if (n <= 1) {
            result = false;
        }
        else {
            // We only need to check up to the sqrt of the number.
            for (long i=2; i*i<=n; i++) {
                if (n % i == 0) {
                // This number is evenly divisible by i, so it is not prime.
                result = false;
                break;
                }
            }
        }
    
        return result;
    }

    /**
    * @param {number} n
    * @return {number}
    */
    public long countPrimes(long n) {
        long result = 0;
    
        for (long i=0; i<n; i++) {
            if (isPrime(i)) {
                result++;
            }
        }
    
        return result;
    }
    
    @RequestMapping(value = "/prime", produces = "application/json")
    public String prime() throws JSONException {
        Long i = countPrimes(10000000);
        return new JSONObject().put("status", "done").put("result", i.toString()).toString();
    }
    
    @RequestMapping(value = "/countries", produces = "application/json")
    @Transactional
    public List<Country> countries() throws JSONException {
    	List<Country> data = (List<Country>) sessionFactory.getCurrentSession()
    			.createCriteria(Country.class)
    			.list();
    	return data;
    }
    
    @RequestMapping(value = "/users", produces = "application/json")
    @Transactional
    public List<User> users() throws JSONException {
    	List<User> data;
            data = (List<User>) sessionFactory.getCurrentSession()
                    //.createSQLQuery("SELECT id,firstName,lastName FROM benchmark.User  join (UserCountryMapping) on (UserCountryMapping.userId=User.id and UserCountryMapping.countryId=:countryID)").setParameter("countryID", 17)
                    .createCriteria(User.class)
                    .createAlias("countries", "countriesAlias")
                    .add(Restrictions.eq("countriesAlias.id", 17))
                    .list();
                
                
    	return data.subList(0, 100);
    }
}
